src/lsst/rsp/catalog.py

```diff
@@ -66,3 +68,23 @@ def retrieve_query(query_url: str) -> pyvo.dal.AsyncTAPJob:
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", category=UserWarning)
         return pyvo.dal.AsyncTAPJob(query_url, session=get_pyvo_auth())
+
+
+async def get_query_history(n: int | None = None) -> list[str]:
+    """Retrieve last n query jobref ids.  If n is not specified, or n<1,
+    retrieve all query jobref ids.
+    """
+    client = RSPClient("/api/tap")
+    params = {}
+    if n and n > 0:
+        params = {"last": f"{n}"}
+    full_history_xml = await client.get("async", params=params)
+    history_dict = xmltodict.parse(full_history_xml.text)
+    joblist = history_dict["uws:jobs"]["uws:jobref"]
+    jobs: list[str] = []
+    for job in joblist:
+        for k, v in job.items():
+            if k == "@id":
+                jobs.append(v)
+                break
```

Comment:
Perhaps slightly more efficient to do a list comprehension here with a direct lookup for each job (untested):
```suggestion
[job["@id"] for job in joblist if "@id" in job]
```

Reply:
Hmmm.  Getting all the information out of the jobs up front would certainly make the extension easier (since right now there's a separate step to get the query text), but on the other hand, it would mean we'd need an API call per job, rather than a single one to get all jobs.

Reply:
Yeah, that's the problem: a single entry is:

```
<uws:jobref id="idw8x0xiyp7hcs24">
<uws:phase>COMPLETED</uws:phase>
<uws:runId>dp02_dc2_catalogs.Object - data-dev</uws:runId>
<uws:ownerId>adam</uws:ownerId>
<uws:creationTime>2025-01-21T19:41:32.860Z</uws:creationTime>
</uws:jobref>
```

So I could get ... well, creation time and phase are somewhat useful but not to my immediate use-case.  Owner ID is useless because it's the same as the owner of the token we used to make the request, and runID isn't all that interesting either.

I want to avoid doing an API call per jobref ID here, and I go to some lengths to cache it when I need to do that in the back-end service in the extension.

Reply:
I guess one option is that some of those fields that are only accessible after an extra call could be properties that will call out to the individual call when accessed by the user?

Also note that we may end up actually showing the query text in the job list. This is being discussed within the IVOA, I brought it up in that meeting and most people were happy with it.

Reply:
Also runId _may_ end up being useful, I think Firefly folks have started using it and they note the table that was accessed & the environment. But also my understanding is that the user will be able to modify this, so it may be something useful for them.

Overall I think happy with you to go with whatever you think works best for now and we can revisit later after getting some user feedback?